src/main/java/org/opensearch/timeseries/model/Config.java

```diff
@@ -0,0 +1,550 @@
+    private static final Logger logger = LogManager.getLogger(Config.class);
+
+    public static final int MAX_RESULT_INDEX_NAME_SIZE = 255;
+    // OS doesn’t allow uppercase: https://tinyurl.com/yse2xdbx
+    public static final String RESULT_INDEX_NAME_PATTERN = "[a-z0-9_-]+";
+
+    public static final String NO_ID = "";
+    public static final String TIMEOUT = "timeout";
+    public static final String GENERAL_SETTINGS = "general_settings";
+    public static final String AGGREGATION = "aggregation_issue";
+
+    // field in JSON representation
+    public static final String NAME_FIELD = "name";
+    public static final String DESCRIPTION_FIELD = "description";
+    public static final String TIMEFIELD_FIELD = "time_field";
+    public static final String INDICES_FIELD = "indices";
+    public static final String UI_METADATA_FIELD = "ui_metadata";
+    public static final String FILTER_QUERY_FIELD = "filter_query";
+    public static final String FEATURE_ATTRIBUTES_FIELD = "feature_attributes";
+    public static final String WINDOW_DELAY_FIELD = "window_delay";
+    public static final String SHINGLE_SIZE_FIELD = "shingle_size";
+    public static final String LAST_UPDATE_TIME_FIELD = "last_update_time";
+    public static final String CATEGORY_FIELD = "category_field";
+    public static final String USER_FIELD = "user";
+    public static final String RESULT_INDEX_FIELD = "result_index";
+    public static final String IMPUTATION_OPTION_FIELD = "imputation_option";
+
+    private static final Imputer zeroImputer;
+    private static final Imputer previousImputer;
+    private static final Imputer linearImputer;
+    private static final Imputer linearImputerIntegerSensitive;
+
+    protected String id;
+    protected Long version;
+    protected String name;
+    protected String description;
+    protected String timeField;
+    protected List<String> indices;
+    protected List<Feature> featureAttributes;
+    protected QueryBuilder filterQuery;
+    protected TimeConfiguration interval;
+    protected TimeConfiguration windowDelay;
+    protected Integer shingleSize;
+    protected String customResultIndex;
+    protected Map<String, Object> uiMetadata;
+    protected Integer schemaVersion;
+    protected Instant lastUpdateTime;
+    protected List<String> categoryFields;
+    protected User user;
+    protected ImputationOption imputationOption;
+
+    // validation error
+    protected String errorMessage;
+    protected ValidationIssueType issueType;
+
+    protected Imputer imputer;
+
+    public static String INVALID_RESULT_INDEX_NAME_SIZE = "Result index name size must contains less than "
+        + MAX_RESULT_INDEX_NAME_SIZE
+        + " characters";
+
+    static {
+        zeroImputer = new ZeroImputer();
+        previousImputer = new PreviousValueImputer();
+        linearImputer = new LinearUniformImputer(false);
+        linearImputerIntegerSensitive = new LinearUniformImputer(true);
+    }
+
+    protected Config(
```

Comment:
What are your plans for being the main distinguishing point between forecaster and detector in structure, should we utilize the difference in interval name b/w the two? We might have covered this index design and I forgot. In the past we talked about having a field to distinguish if forecaster or detector for frontend.